core-bundle/src/Cron/Cron.php

```diff
@@ -135,11 +137,40 @@ private function doRun(array $cronJobs, string $scope, bool $force = false): voi
             $repository->unlockTable();
         }
 
-        // Execute all cron jobs to be run
-        foreach ($cronJobsToBeRun as $cron) {
+        $this->executeCrons($cronJobsToBeRun, $scope);
+    }
+
+    /**
+     * @param array<CronJob> $crons
+     */
+    private function executeCrons(array $crons, string $scope): void
+    {
+        /** @var array<string, PromiseInterface> $promises */
+        $promises = [];
+
+        foreach ($crons as $cron) {
             $this->logger?->debug(sprintf('Executing cron job "%s"', $cron->getName()));
 
-            $cron($scope);
+            $promise = $cron($scope);
+
+            if ($promise instanceof PromiseInterface) {
+                $promise->then(
+                    function () use ($cron): void {
+                        $this->logger?->debug(sprintf('Asynchronous cron job "%s" finished successfully', $cron->getName()));
+                    },
+                    function ($reason) use ($cron): void {
+                        $this->logger?->debug(sprintf('Asynchronous cron job "%s" failed: %s', $cron->getName(), $reason));
```

Comment:
```suggestion
                        $this->logger?->debug(sprintf('Asynchronous cron job "%s" failed: %s', $cron->getName(), (string) $reason));
```
AFAIK the reason is of type `mixed` an can be an exception object for example.

Reply:
I actually did add a `(string)` cast but phpstorm tells me it's obsolete due to `sprintf()` in conjuction with `%s` which I guess does that anyway? ;)